### ISSUE TITLE
VS Code doesn't work on Fedora 35/Wayland/Nvidia

### ISSUE THREAD

**Issue by freevatar** [microsoft/vscode#134255]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version:
`1.61.0-insider`
83b7d33abba1fbc9ba6dbf25b53b47821086b22f
x64

- OS Version: `Fedora 35`

Steps to Reproduce:

1. Install VS Code Insiders Edition
2. Start

How reproducible:
* Always

Actual results:
* VS Code doesn't work on Fedora 35/Wayland/Nvidia. It shows completely blank white window with title bar reads `Get Started - Visual Studio Code`

Additional info:
* works on Xorg.
* works on Wayland with `--no-sandbox` flag set but laggy
* works on Wayland with `--disable-gpu` flag set
* works on Wayland with `-enable-features=UseOzonePlatform --ozone-platform=wayland` flags set but the window doesn't have title bar in this case due to [issue](https://github.com/electron/electron/issues/27522)
* similar issue: [issue](https://github.com/signalapp/Signal-Desktop/issues/5361)

Software Details:

- kernel: `5.14.7-300`
- Nvidia driver:  `470.74`
- gnome-shell: `41.0-1`
- glibc: `2.34-6`
- xorg-x11-server-Xwayland: `21.1.2-2`
- egl-wayland: `1.1.7-2`
- libwayland: `1.19.0-2`

**Comment by kaushalyap** [comment]
@jrieken following works

```
code --disable-gpu-sandbox
```

**Comment by deepak1556** [comment]
Can you provide the output of `code --verbose` for wayland scenario.

**Comment by freevatar** [comment]
Here's the outpupt:
[vscode-wayland.log](https://github.com/microsoft/vscode/files/7270474/vscode-wayland.log)

I guess main line here:
`libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)`

**Comment by larsks** [comment]
I have the same failures on Fedora 34, where the output of `--verbose` fails with:

```
[2622212:1005/164712.299439:ERROR:gpu_process_host.cc(1007)] GPU process exited unexpectedly: exit_code=159
[2622212:1005/164712.299453:WARNING:gpu_process_host.cc(1321)] The GPU process has crashed 9 time(s)
[2622212:1005/164712.299461:FATAL:gpu_data_manager_impl_private.cc(415)] GPU process isn't usable. Goodbye.
```

This is caused by the fact that recent versions of glibc default to using the `clone3` system call, which in `code` (and a variety of other chromium-based apps) isn't permitted by the seccomp filter. Running `strace` reveals:

```
2620574 --- SIGSYS {si_signo=SIGSYS, si_code=SYS_SECCOMP, si_errno=EPERM, si_call_addr=0x7f65ea6e92ed, si_syscall=__NR_clone3, si_arch=AUDIT_ARCH_X86_64} ---
```

The workaround described here (`--disable-gpu-sandbox`) works on Fedora 34; a proper fix is to update the sandbox to permit calls to `clone3`.

**Comment by deepak1556** [comment]
@larsks the issue you are facing has been addressed in current insiders via https://github.com/microsoft/vscode/issues/133593 and the fix will be available in upcoming `1.61` stable release.

**Comment by deepak1556** [comment]
@freevatar can you check if the issue repros with `1.61` stable release ?

**Comment by vchernin** [comment]
Seems like an upstream bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1254483

**Comment by freevatar** [comment]
They might be related.

`Chrome` shows completely transparent window

`VS Code` windows is non-transparent and has title bar
![Screenshot from 2021-10-13 01-38-07](https://user-images.githubusercontent.com/10532243/137073800-5d72566a-bd68-4128-846a-ecdf49afdd38.png)

**Comment by Schmensch** [comment]
You can use the ozone workaround to have native Wayland support and then enable Code's own window controls. It isn’t prefect but it works.

**Comment by freevatar** [comment]
When I run VS Code with `--use-gl=egl` flag, it works with Nvidia driver `495.44`.

**Comment by adyomin** [comment]
Very similar issues with VSCode, Chrome, Chromium, Brave as described by the OP and @freevatar

Fedora 35
Nvidia 495.44
code 1.61.2 
org.chromium.Chromium 95.0.4638.69
Gnome 41 Wayland session

**Comment by freevatar** [comment]
@adyomin
The current status is `Chrome` works for me but `VS Code` doesn't.

**Comment by Loki-Afro** [comment]
i do not use VS Code at all. but having the same issue with other electron apps.
- Rocketchat
- Lens

also on fedora 35, worked fine on 34.
My upgrade path was a bit odd: i am using the properitary nvidia driver from rpm fusion, apparently my gpu reached now legacy state, so i had to uninstall my nvidia drivers and switch to a legacy package. I thought that might be the cause of the issue, that some magic thing is now missing.

On the other hand, i see that this problem pops up every once in a while and the causes were glibc was updated and was incompatible somehow (2016) - that might be a fix then on chromium side.

however, what is definitely strange spotify still works - maybe there is a hack involved, but its also electron crap.
And even more weirder: chromium and google-chrome are still functional. Which makes me think that chromium is not the issue here?

```
zarathustra@wotan ~ % google-chrome --version
Google Chrome 96.0.4664.45 
zarathustra@wotan ~ % chromium-browser --version
Chromium 94.0.4606.81 Fedora Project
zarathustra@wotan ~ % screenfetch
           /:-------------:\          zarathustra@wotan
        :-------------------::        OS: Fedora 
      :-----------/shhOHbmp---:\      Kernel: x86_64 Linux 5.14.18-300.fc35.x86_64
    /-----------omMMMNNNMMD  ---:     Uptime: 1d 5h 23m
   :-----------sMMMMNMNMP.    ---:    Packages: 4106
  :-----------:MMMdP-------    ---\   Shell: zsh 5.8
 ,------------:MMMd--------    ---:   Resolution: 5760x1080
 :------------:MMMd-------    .---:   DE: LXDE
 :----    oNMMMMMMMMMNho     .----:   WM: OpenBox
 :--     .+shhhMMMmhhy++   .------/   WM Theme: Onyx
 :-    -------:MMMd--------------:    GTK Theme: BlackMATE [GTK2]
 :-   --------/MMMd-------------;     Icon Theme: Fedora
 :-    ------/hMMMy------------:      Font: Sans 10
 :-- :dMNdhhdNMMNo------------;       Disk: 247G / 362G (69%)
 :---:sdNMMMMNds:------------:        CPU: Intel Core i7-6700HQ @ 8x 3.5GHz [51.0°C]
 :------:://:-------------::          GPU: NVIDIA GeForce GTX 965M/PCIe/SSE2
 :---------------------://            RAM: 11593MiB / 32043MiB                                 
```

**Comment by vchernin** [comment]
Perhaps Lens and Rocketchat are using older Electron/Chromium versions, and the issue is solved in newer Electron/Chromium builds (which Spotify uses).

**Comment by ZanderBrown** [comment]
> however, what is definitely strange spotify still works - maybe there is a hack involved, but its also electron crap.

It's not

**Comment by Loki-Afro** [comment]
@ZanderBrown dunno how you installed spotify, but i choose that weird path with lpf that was suggested 2 years ago or so - maybe thats why it works on my box. https://docs.fedoraproject.org/en-US/quick-docs/installing-spotify/#_using_the_rpm_fusion_repository

**Comment by ZanderBrown** [comment]
I don't use Spotify, I was simply noting it doesn't use Electron

**Comment by StefanoD** [comment]
I can confirm this bug. I'm also using Fedora 35, Wayland and Nvidia driver. 495.44

What works for me is following: `code --disable-gpu`

**Comment by RickyC0626** [comment]
On Fedora 35, [updating to the latest version of VS Code](https://code.visualstudio.com/docs/setup/linux#_rhel-fedora-and-centos-based-distributions) fixed the problem for me.

**Comment by freevatar** [comment]
> On Fedora 35, [updating to the latest version of VS Code](https://code.visualstudio.com/docs/setup/linux#_rhel-fedora-and-centos-based-distributions) fixed the problem for me, so it seems like as of `1.63.2` this problem has been fixed.

No, it doesn't.

**Comment by RickyC0626** [comment]
> > On Fedora 35, [updating to the latest version of VS Code](https://code.visualstudio.com/docs/setup/linux#_rhel-fedora-and-centos-based-distributions) fixed the problem for me, so it seems like as of `1.63.2` this problem has been fixed.
> 
> No, it doesn't.

Oh, then I was wrong. It seems to be a different problem from what I had when updating from vscode `v1.60` to `v1.63`. On `v1.60`, the app would close immediately after startup. I also had the latest Nvidia driver `495.xx`.

**Comment by m0xxz** [comment]
The `--disable-gpu-sandbox` work around is my go to right now for any electron based application.

**Comment by struq** [comment]
`--enable-features=Vulkan` also works for me and has better performance than `--disable-gpu`

**Comment by ethan-halsall** [comment]
> `--enable-features=Vulkan` also works for me and has better performance than `--disable-gpu`

This works and gets great performance, far better than the ozone workaround, thanks!

**Comment by krshrimali** [comment]
Hi, has there been any progress on this? There are lots of comments, so maybe I missed some context, but @deepak1556 - it would be helpful if you can help get this resolved as Fedora 36 releases soon.

Also, `--enable-features=Vulkan` is a work-around that works, but any idea on how to set it as default in launch options for VSCode?

**Comment by deepak1556** [comment]
This should be addressed once we update to Electron version that contains fix for https://github.com/electron/electron/issues/32436

**Comment by krshrimali** [comment]
> This should be addressed once we update to Electron version that contains fix for [electron/electron#32436](https://github.com/electron/electron/issues/32436)

Thank you so much, @deepak1556 - for your response. Just to confirm, does the "April 2022" milestone goes out towards the end of April? Or the first week?

**Comment by deepak1556** [comment]
It would be End of April
